### PR TITLE
Adding code to not allow importing from registry when contenType is not kubevirt

### LINKED ISSUE #808: The check not to allow import from registry if contentType is not 'kubevirt' was removed
/kind bug

**What happened**:
The check not to allow import from registry if contentType is not 'kubevirt' was removed

**What you expected to happen**:
getting error when trying to import from registry with contentType = archive. Instead the import is completed.
**How to reproduce it (as minimally and precisely as possible)**:

### PR DESCRIPTION
**Pull request by zvikorn**
Signed-off-by: tavni <tavni@redhat.com>

**What this PR does / why we need it**:
fixing a bug where we allow import from registry when contentType is not kubevirt
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #808 

**Release note**:

```release-note
BugFix: The check not to allow import from registry if contentType is not 'kubevirt' was removed #808 
```

### CODE REVIEW

**Comment by zvikorn** [comment]
Done

File: pkg/apiserver/webhooks/validating-webhook/validating-webhook.go [hunk 1]
```diff
@@ -147,6 +147,16 @@ func validateDataVolumeSpec(field *k8sfield.Path, spec *cdicorev1alpha1.DataVolu
 		return causes
 	}
 
+	if spec.Source.Registry != nil && spec.ContentType != "" && string(spec.ContentType) != string(cdicorev1alpha1.DataVolumeKubeVirt) {
+		sourceType = field.Child("contentType").String()
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("ContentType must be KubeVirt when SourceType is Registry"),
```
Comment (awels) [inline pkg/apiserver/webhooks/validating-webhook/validating-webhook.go]: "ContentType must be KubeVirt when Source is Registry"

Comment (zvikorn) [reply pkg/apiserver/webhooks/validating-webhook/validating-webhook.go]: I think this is what we decided. For now, I do not believe we support special 'archive' case for registry.

Comment (zvikorn) [reply pkg/apiserver/webhooks/validating-webhook/validating-webhook.go]: :-) done

**Pull request merged**